Makefile

```diff
@@ -148,6 +148,13 @@ binary-numacell: build-tools
 		-ldflags "-s -w" \
 		test/deviceplugin/cmd/numacell/main.go
 
+binary-nrocli: build-tools
+	CGO_ENABLED=0 go build \
+		-mod=vendor \
+		-o bin/nrocli \
+		-ldflags "-s -w" \
+		cmd/numaresourcesoperator/main.go
```

Comment:
this is actually misleading as `main.go` of the operator proper sits in the root directory - `operator-sdk` mandated layout. Let's rename the tool `nrocli` so ti reads like `cmd/nrocli/main.go`.
On top of that

- do we see more use cases coming that suggests we need more than a validation tool but a generic command line tool?
- we may want to follow the `rte` footsteps and just have something like `nrocli/main.go` (drop `cmd`)

Reply:
Agree with the naming issue.

Well in my point of view NRO is a complex component with lot of elements and I think in the near future we could make use of some cli tools in the near future to ease testing or even for the people installing/troubleshooting in the wild to be able to make checks in the cluster before installation or get information while it is running .. .and the amount of code infra needed is minimum and it is kept in a separate folder so does not mess up with the operator code 

Regarding following the `rte` "path ... I do not really get the difference right now.

Reply:
I don't really see these usecases clearly, maybe we can clarify them before?
I don' really have any issue with the `cmd/whatever/main.go` - actually I like this layout much more, but having three different layouts will be a problem:

1. operator with `main.go` in the root directory (mandated by operator-sdk)
2. RTE bundled inside `rte` - because we don't want to mix packages here
3. `nrocli` using the standard `cmd/nrocli/main.go` pattern

let's keep the variations to a minimum. Here's why I suggested a simple `nrocli/main.go`. In this case the packages of `nrocli` and the operator CAN mix (which is also true to some extent for RTE and the operator, but that's trickier).

Reply:
Ok, I will change it.